internal/logger/logger.go

```diff
@@ -0,0 +1,249 @@
+// Originally source from https://gist.github.com/panta/2530672ca641d953ae452ecb5ef79d7d
+// Modified logging package to include more context and Level to be configured
+// duplicating the method: https://github.com/learning-cloud-native-go/myapp/blob/step-6/util/logger/logger.go
+package logger
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+type Logger struct {
+	*zerolog.Logger
+}
+
+var Log *Logger //nolint: gochecknoglobals
+// Configuration for logging.
+type Config struct {
+	// Set Enable to false to disable logging
+	Enable bool
+	// Enable console logging
+	ConsoleLoggingEnabled bool
+	// EncodeLogsAsJson makes the log framework log JSON
+	EncodeLogsAsJson bool
+	// FileLoggingEnabled makes the framework log to a file
+	// the fields below can be skipped if this value is false!
+	FileLoggingEnabled bool
+	// Directory to log to to when filelogging is enabled
+	Directory string
+	// Filename is the name of the logfile which will be placed inside the directory
+	Filename string
+	// MaxSize the max size in MB of the logfile before it's rolled
+	MaxSize int
+	// MaxBackups the max number of rolled files to keep
+	MaxBackups int
+	// MaxAge the max age in days to keep a logfile
+	MaxAge int
+	// Level is either info or debug output level
+	Level string
+}
+
+// init logger with disabled state on import to avoid any nil pointers if unused.
+func init() { //nolint:gochecknoinits
```

Comment:
First, I included this to eliminate nil pointer issues in tests by initializing as disabled logger. See line 78